.github/workflows/docs-publish.yml

```diff
@@ -0,0 +1,51 @@
+name: Build and Publish Documentation
+
+on:
+  push:
+    branches:
+      - main
```

Comment:
We should probably only trigger on release events (similar to `docker-build.yml` and `publish-pypi.yml`). Otherwise the public APIs may become out-of-sync with the latest release.